### DISCUSSION TITLE
Get USB Camera Current Settings Action

### DISCUSSION THREAD

**Discussion by UUoocl** [WarmUpTill/SceneSwitcher#705] [Ideas]
Hello,
I’d like to suggest getting a Video Capture Device’s settings as a macro Condition and Action.  
For example, the [Insta360 Link camera](https://www.insta360.com/product/insta360-link) uses AI to track a subject.  As the camera pans and tilts it’s position values could be used as a macro condition and action.   
The Move Transition plugin has a Move Video Capture Device option which can read the camera settings.  There is a “Get Value” button that returns the camera’s setting in detail.  
_Move Transition Filter settings._ 
![image](https://user-images.githubusercontent.com/99063397/228726723-09062e37-a2fe-4129-bd18-3dcba6df0c5f.png)

Could the Advanced Scene Switcher have a condition that checks if the camera position settings changed and an action that gets the current camera position settings?
_Advanced Scene Switcher reading the Move Transition Filter_
![image](https://user-images.githubusercontent.com/99063397/228726820-e1f0db48-4ee0-44ca-a6eb-f9f2516756a7.png)

Thank you

**Comment by WarmUpTill** [comment]
Thanks for the suggestion - I will look into it!

**Reply by UUoocl** [reply]
Thanks for responding to this idea. 

The [uvc-util](https://github.com/jtfrey/uvc-util) is a command-line utility on Mac OS X that can query and modify USB Video Capture camera controls.  Do you think the uvc-util could be added to Advanced Scene Switcher to create camera conditions and actions?

I'm using uvc-util commands in a [Lua script](https://github.com/UUoocl/Moving_Virtual_Background) to get the position of a point, Tilt, zoom (PTZ) camera. 
I couldn't figure out how to use Advanced Scene Switcher to Run the uvc-util commands. Can Advanced Scene Switcher run command line programs in MacOS?

**Reply by WarmUpTill** [reply]
In the meantime it might already be possible to read out the setting value using the "Source" condition type.

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/96ae12d9-df4d-4bea-88c2-e6078c68309a)

(This is assuming they are available to OBS via the source settings dialog as well)

> Can Advanced Scene Switcher run command line programs in MacOS?

You should be able to run any binary you like using the "Run" action type.
For example:

![image](https://github.com/WarmUpTill/SceneSwitcher/assets/19472752/8c580439-8f81-4d73-ae8b-1246a7c5eebf)

It has to be noted however that you _might_ have to configure some Security settings on MacOS so that such operations are allowed.

**Reply by UUoocl** [reply]
Thank you for responding to my questions.  The things I've tried to open the uvc-util app using the Run Condition have been unsuccessful.
I've tried clearing and deleting attributes with xattr for Advanced Scene Switcher and uvc-util.   
Would you have suggestions on what to try next?
![image](https://github.com/WarmUpTill/SceneSwitcher/assets/99063397/7a514085-63a5-484f-bdd2-622753a3cdcc)
